### DISCUSSION TITLE
How reasonable for raytracing

### DISCUSSION THREAD

**Discussion by lonnietc** [MethanePowered/MethaneKit#113] [Q&A] [answered]
Hello,

I am just now getting involved with upgrading an older engine and I really like what I have seen in the MethaneKit so far as it seems much faster than other frameworks like BGFX and Diligent that I have also looked at recently.

Also, I am greatly interested in real-time Raytracing efforts since I think that they are the real future to 3D engines. 

With that in mind, I was wondering how applicable, or how hard it might be to use MethaneKit in a raytracing application or example since you currently have some many really nice examples so far.

Wanted to get your thoughts on this.
Thanks

**Comment by egorodet** [comment, marked as answer]
Hello @lonnietc,

I'm glad that you liked Methane Kit and preferred it instead of BGFX and Diligent Engine. But I must admire that these frameworks are in much more mature and stable state, which allows to use them for game engine development. Methane Kit is still being actively developed, has frequently changing API and a lot of required features still to be done.

I'd really like also to add support of Ray-Tracing pipeline in future, but it's not a part of my near term plans, because there are more important features that need to be added first.

**Reply by lonnietc** [reply]
Thanks for the discussion on this and I will have to keep an eye out for the Methane Kit development since it does have many nice features and a high framerate which I need for the updating of the 3D game engine which will also be experimental for a good while as well.  I will also look forward to ray tracing implementations in Methane Kit and maybe we can discuss it more going forward at some point.
